File: app/src/main/java/com/narcyber/mvpbasics/network/WeatherApiClient.java
package com.narcyber.mvpbasics.network;

import com.narcyber.mvpbasics.helper.ConstantHelper;

import retrofit2.Retrofit;
import retrofit2.adapter.rxjava3.RxJava3CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

public class WeatherApiClient {
    private static Retrofit retrofit = null;

    public static Retrofit getClient() {
        if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .addConverterFactory(GsonConverterFactory.create())
                    .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
                    .baseUrl(ConstantHelper.BASE_URL)
                    .build();
        }
        return retrofit;
    }
}

File: app/src/main/java/com/narcyber/mvpbasics/network/WeatherApiClient.kt
package com.narcyber.mvpbasics.network

import com.narcyber.mvpbasics.helper.ConstantHelper
import retrofit2.Retrofit
import retrofit2.adapter.rxjava3.RxJava3CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

object WeatherApiClient {
    val weatherClient: Retrofit by lazy {
        Retrofit.Builder()
            .addConverterFactory(GsonConverterFactory.create())
            .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
            .baseUrl(ConstantHelper.BASE_URL)
            .build()
    }

}